README.md

```diff
@@ -74,7 +74,11 @@ Error: Resource not accessible by integration
 
 ```yml
 name: CI
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
```

Comment:
`main` instead of `master`.
Or place both.

Reply:
Hey @climba03003, thanks for pointing this out. I will update the branch.
Leaving `Basic Example` as it is may be misleading as the actions used in the composite one will fail if the filter on push is not properly set.

Reply:
`Basic Example` means to be basic usage. Not every need to filter out the `push` event and I don't see the need to filter out by default.

Reply:
@climba03003 it is necessary because this action doesn't work on pushes, so filtering is necessary and should be in the basic example

Reply:
If it is requirement for not running on `pull`. It should add a `if` cause for the `step`.
The changes here is **NOT** don't run on `pull`, but only run on `master`.

```
    if: >
        github.event_name == 'pull_request' &&
        github.event.pull_request.user.login == 'dependabot[bot]'
```

Reply:
We intentionally avoided the if on the step to keep the usage of the action simple. A CI workflow anyway is better configured the way it is here, because if you don't filter the push trigger you'll always have duplicate runs, which is not what you usually want.

Reply:
Anyway, I don't block it. But it's weird for the reason.

If you want to keep it simple, then no exception. And not everyone wish to prevent the `push` trigger.
It even doesn't hurt if you run multiple times (ensure no flaky test).

Reply:
I see your point, we used to have that in the past. Now for another couple of reasons it may make sense to reintroduce this condition on the step itself, but I think we can handle it even better within the action itself, which we aren't at the moment.

Reply:
@simoneb
I see the issue on your end and the update on readme will not solve your issue. It will throw error even with the filter because it runs on `pull` event.

The proper fix should be update the `action.yml`, so `dependabot/fetch-metadata` will runs on PR only. It is the one causing problem.